src/LayerEditor/ui/data/le_model.py

```diff
@@ -587,3 +555,23 @@ def change_curr_block(self, block=None):
             old_block = self.gui_block_selector.value
             self.gui_block_selector.value = block
             self.active_block_changed.emit(old_block.id)
+
+    def add_shape_file(self, filename):
+        if not self.shapes_model.is_file_open(filename):
+            shp_item = self.shapes_model.add_file(filename)
+            return shp_item.errors
+        return None
+
+    def decompositions_empty(self, ignore_shapes=False, ignore_surf=False, ignore_decomps=False):
```

Comment:
Add doc string.

Reply:
Renamed to `le_model_empty`
Added:
"""Return True if LEModel is empty, or if it only contains objects which are ignored. Otherwise returns False."""